### ISSUE TITLE
Error when importing JupiterMag

### ISSUE THREAD

**Issue by nkruegler** [mattkjames7/JupiterMag#3]
I am currently unable to import the JupiterMag package in python after installing with pip. The package appears to install with no errors, but the dll files do not appear to be created during installation with pip, which throws errors when attempting to import. Here is the output given by iPython when calling `import JupiterMag`:
![image](https://github.com/mattkjames7/JupiterMag/assets/144250047/e4170d05-acf8-4b3c-a6f3-030cca96987b)
![image](https://github.com/mattkjames7/JupiterMag/assets/144250047/0823dfba-bd93-445e-9ed3-cbaa8c9270a0)

I used this package frequently last summer, but have since redownloaded my installation of Anaconda and made other changes to help another python package work, so am unsure what changes I may have done that would result in this error.

I verified in the command prompt that I have `g++` and `ld` installed. I tried the alternative methods mentioned in the various documentation (and earlier versions of JupiterMag), which resulted in similar errors that mentioned the missing dll files.

For reference, I am using a laptop running Windows 10. I use Anaconda as my python package manager, but I have also found the same errors using a base python installation.

Any help you could provide with this would be appreciated. I you need any other info, please let me know. This is a great package and has been super useful to me so far with my research.

**Comment by mattkjames7** [comment]
Hi nkruegler,

I have recently helped another researcher with what appears to be the same problem. I haven't yet figured out what cause the compile to fail (but the pip install to appear successful), but I have found a couple of ways around this. 

The easiest way (assuming it will work for you, and you are using a 64-bit Windows and an x86 processor) would be to uninstall the existing package and download and install this: [https://github.com/mattkjames7/JupiterMag/releases/download/v1.3.0/JupiterMag-1.3.0-py3-none-win_amd64.whl](https://github.com/mattkjames7/JupiterMag/releases/download/v1.3.0/JupiterMag-1.3.0-py3-none-win_amd64.whl)

From an anaconda powershell/prompt, with the above package in your working directory, install using:

`pip install JupiterMag-1.3.0-py3-none-win_amd64.whl`

With a bit of luck, the precompiled DLL within that package will import without any problems.

The less easy way involves navigating to the directory where the C++ library should be in an anaconda powershell/prompt, which in your case appears to be `~\anaconda3\envs\test\Lib\site-packages\JupiterMag\__data\libjupitermag`

When you're in that directory, run `./compile.bat` - there would be a few warnings, but it should compile fine unless any of the build tools are missing. If it works, a new DLL should have appeared within the `\lib` subdirectory.

I hope that's helpful, please let me know if any of the above fixes the problem or not. If it continues to not work, we can dig deeper!

Thanks for using my code!

**Comment by nkruegler** [comment]
I went the route of running the `compile.bat` script, which seems to have worked! There was an identical issue with the DateTimeTools package that JupiterMag relies on, so I had to fix that first. I used the same method and ran the `compile.dat` for that package. I was able to import both separately in python, and the JupiterMag package now imports properly. 

DateTimeTools appears to be a C++ wrapper for Python like JupiterMag, so it seems like the issue is shared between the two.

Thank you so much for your help! I appreciate the quick response.
